Modules/socketmodule.c

```diff
@@ -1493,9 +1493,18 @@ makesockaddr(SOCKET_T sockfd, struct sockaddr *addr, size_t addrlen, int proto)
             PyObject *addrobj = makebdaddr(&_BT_L2_MEMB(a, bdaddr));
             PyObject *ret = NULL;
             if (addrobj) {
-                ret = Py_BuildValue("Oi",
-                                    addrobj,
-                                    _BT_L2_MEMB(a, psm));
+                /* Retain old format for non-LE address */
+                if (_BT_L2_MEMB(a, bdaddr_type) == BDADDR_BREDR) {
+                    ret = Py_BuildValue("Oi",
+                                        addrobj,
+                                        _BT_L2_MEMB(a, psm));
+                } else {
+                    ret = Py_BuildValue("OiiB",
+                                        addrobj,
+                                        _BT_L2_MEMB(a, psm),
+                                        _BT_L2_MEMB(a, cid),
+                                        _BT_L2_MEMB(a, bdaddr_type));
```

Comment:
A tip, this can reduce one nesting and an assignment statement. WDYT?
```suggestion
if (addrobj) {
      /* Retain old format for non-LE address */
      ret = _BT_L2_MEMB(a, bdaddr_type) == BDADDR_BREDR
		    ? Py_BuildValue("Oi", addrobj, _BT_L2_MEMB(a, psm))
		    : Py_BuildValue("OiiB", addrobj,
					                _BT_L2_MEMB(a, psm),
					                _BT_L2_MEMB(a, cid),
					                _BT_L2_MEMB(a, bdaddr_type));
}
```

Reply:
Thank you for the suggestion! It is indeed more compact, however I personally also find it less readable. I'll let the CPython reviewers decide if your suggestion is more inline with their desired code style.

Reply:
IMO the expanded one is more readable, especially if we need to add new cases later.

As someone who's not familiar with these structs, I see a missing case when `bdaddr_type` is `BDADDR_BREDR` but `cid` is set. Could you add a comment if that's meaningless, or an assert if it's impossible?